Assignment Day 2 _9th December 2020.ipynb

```diff
@@ -24,7 +24,7 @@
     {
      "data": {
       "text/plain": [
-       "'Mary had a little lamb'"
+       "'Mary had a little lamb and also a girl with her'"
```

Comment:
**Performance Issue**: The change in the string to include 'and also a girl with her' increases the length of the data being processed. This modification, while seemingly minor, can have implications on memory usage and potentially on the execution time, especially if this string manipulation is part of a larger, more complex data processing operation or if it's executed multiple times within a loop. <br> **Fix**: Evaluate the necessity of the string modification. If the added information is crucial, consider optimizing the string storage or processing methods. For example, if this operation is within a loop or a frequently called function, ensure that the string manipulation is optimized for performance. This could involve using more efficient string handling techniques or ensuring that the operation is only performed when absolutely necessary. <br> **Code Suggestion**: 
 ```
 Evaluate the necessity of the string modification. If crucial, optimize string handling:

 - Before modification:
   text = 'Mary had a little lamb'

 - After modification, if necessary:
   text = 'Mary had a little lamb and also a girl with her'
   # Ensure optimization for memory and performance, especially within loops or frequent calls.
 ```